common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -53,6 +53,38 @@ def get_queryset(self):
         # to Admin users
         return CourseEntitlement.objects.all().select_related('user').select_related('enrollment_course_run')
 
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        entitlement = serializer.instance
+        user = entitlement.user
+
+        # find all course_runs within the course
+        course_runs = get_course_runs_for_course(entitlement.course_uuid)
+
+        # check if the user has enrollments for any of the course_runs
+        user_run_enrollments = [CourseEnrollment.get_enrollment(user, CourseKey.from_string(course_run.get('key'))) for course_run in course_runs if CourseEnrollment.get_enrollment(user, CourseKey.from_string(course_run.get('key')))]
+ 
+        # filter to just enrollments that can be upgraded.
+        upgradeable_enrollments = [enrollment for enrollment in user_run_enrollments if enrollment.upgrade_deadline and enrollment.upgrade_deadline > timezone.now()]
```

Comment:
I think you could consider simplifying this to
```python
filter(lambda e: e.update_deadline and e.upgrade_deadline > timezone.now(), user_run_enrollments)
```
I think it might show the intention of "filtering" slightly better (that's just my opinion though).

Reply:
I would actually prefer this be a look than a composition or a lambda.  I find it to be less clear as both, again an opinion.